openmc/tallies.py

```diff
@@ -1672,7 +1679,11 @@ def _align_tally_data(self, other):
             self_repeat_factor = other.num_nuclides
             other_tile_factor = self.num_nuclides
 
-            # Replicate the data
+            # Tile / repeat the tally data for the tally outer product
```

Comment:
I'll have to revisit the changes to the NumPy array reshaping here when I have a little more time. This is very difficult to generalize, and may be the reason that this PR is failing some of the tests.